src/core/dynamic-user.c

```diff
@@ -770,7 +770,7 @@ int dynamic_creds_acquire(DynamicCreds *creds, Manager *m, const char *user, con
 
                 if (creds->user && (!group || streq_ptr(user, group)))
                         creds->group = dynamic_user_ref(creds->user);
-                else {
+                else if (group) {
```

Comment:
This looks trivially correct, because dynamic_user_acquire will assert that the argument it not null. So let's merge.